java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -311,20 +318,27 @@ public String linkUrl() {
   }
 
   private void addMethodPropertiesToConfigMethodProperties(
-      Set<Map<String, Object>> methodProperties) {
+      Set<Map<String, Object>> methodProperties, boolean isArgumentIndexOptional)
+      throws ParseException {
     for (Map<String, Object> methodProperty : methodProperties) {
-      String flagType = getValueStringFromMap(methodProperty, FLAG_TYPE_KEY);
       String methodName = getValueStringFromMap(methodProperty, METHOD_NAME_KEY);
-      if (flagType != null && methodName != null) {
+      if (methodName != null && getValueStringFromMap(methodProperty, FLAG_TYPE_KEY) != null) {
+        if (!isArgumentIndexOptional && getArgumentIndexFromMap(methodProperty) == null) {
+          throw new ParseException(
+              "methodProperty did not have argumentIndex. Use Piranha:ArgumentIndexOptional if required. Check:"
```

Comment:
This should be a more explanatory error message:

"methodProperty did not have argumentIndex. By default, Piranha requires an argument index for flag APIs, to which the flag name/symbol will be passed. This is to avoid over-deletion of all occurrences of a flag API method. If you are sure you want to delete all instances of the method below, consider using Piranha:ArgumentIndexOptional=true to override this behavior. Check:"